tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/nvptx_backend_lib.cc

```diff
@@ -444,24 +510,80 @@ void GPUBackendInit(const HloModuleConfig& hlo_module_config) {
   InitializePasses(registry);
 }
 
-}  // namespace
+}  // namespace nvptx
+
+// Logic specific to LLVM AMDGPU backend
+namespace amdgpu {
+
+// Gets the ROCm-Device-Libs filenames for a particular AMDGPU version.
+static std::vector<string> GetROCDLPaths(int amdgpu_version,
+                                         const string& rocdl_dir_path) {
+  // AMDGPU version-neutral bitcodes
+  std::vector<string> rocdl_filename_vector{
+      "hc.amdgcn.bc",
+      "opencl.amdgcn.bc",
+      "ocml.amdgcn.bc",
+      "ockl.amdgcn.bc",
+      "oclc_finite_only_off.amdgcn.bc",
+      "oclc_daz_opt_off.amdgcn.bc",
+      "oclc_correctly_rounded_sqrt_on.amdgcn.bc",
+      "oclc_unsafe_math_off.amdgcn.bc"};
+
+  // Construct full path to ROCDL bitcode libraries
+  std::vector<string> result;
+  for (auto& filename : rocdl_filename_vector) {
+    result.push_back(tensorflow::io::JoinPath(rocdl_dir_path, filename));
+  }
+
+  // Add AMDGPU version-specific bitcodes
+  result.push_back(tensorflow::io::JoinPath(
+      rocdl_dir_path, tensorflow::strings::StrCat(
+                          "oclc_isa_version_", amdgpu_version, ".amdgcn.bc")));
+  return std::move(result);
+}
+
+// Links ROCm-Device-Libs into the given module if the module needs it.
+Status LinkROCDLIfNecessary(llvm::Module* module, int amdgpu_version,
+                            const string& rocdl_dir_path) {
+  if (!CouldNeedDeviceBitcode(*module)) {
+    return tensorflow::Status::OK();
+  }
+
+  return LinkWithBitcodeVector(module,
+                               GetROCDLPaths(amdgpu_version, rocdl_dir_path));
+}
+
+}  // namespace amdgpu
 
 StatusOr<string> CompileToPtx(llvm::Module* module,
-                              std::pair<int, int> compute_capability,
+                              GpuVersion gpu_version,
```

Comment:
Functions with "Ptx" in the name are NVidia-specific. Please move it under the nvptx namespace and keep using compute_capability rather than the generic GpuVersion.

Reply:
will do. In such case I'll also put AMDGPU-specific `CompileToHsaco` under `amdgpu` namespace.